File: src/examples/java/src/SampleClient.java
import org.apache.thrift.transport.TTransport;
import org.apache.thrift.transport.THttpClient;
import org.apache.thrift.protocol.TBinaryProtocol;
import org.apache.thrift.protocol.TProtocol;
import java.nio.ByteBuffer;
import datahub.*;

/** 
 * Sample Java Client for DataHub
 * 
 * @author anantb
 * @date 11/07/2013
 * 
 */

public class SampleClient {
  public static void main(String [] args) {
    try {
      // Before running you will need to populate the below variables
      // and create a corresponding repo and table in your datahub acount
      
      String username = "anantb";
      String password = "anant";
      String repo = "test";
      String table = "demo";
      
      // http setup
      TTransport transport = new THttpClient("http://datahub.csail.mit.edu/service");
      TProtocol protocol = new  TBinaryProtocol(transport);
      DataHub.Client client = new DataHub.Client(protocol);

      System.out.println("Version: " + client.get_version());

      // open connection
      ConnectionParams con_params = new ConnectionParams();
      con_params.setUser(username);
      con_params.setPassword(password);
      Connection con = client.open_connection(con_params);
	     
      
      /********************************************************
      CONNECT TO AND QUERY DB USING YOUR USERNAME AND PASSWORD
      ********************************************************/
      // This will initially thrown an exception.
      // You will need to create a username, password,
      // repository, and table to query.
      System.out.println("\n\nTrying to query a user's table...");
      String query = "select * from " + username + "." + repo + "." + table;
      ResultSet res = client.execute_sql(con, query, null);

      // print field names
      for (String field_name : res.getField_names()) {
        System.out.print(field_name + "\t");
      }

      System.out.println();

      // print tuples
      for (Tuple t : res.getTuples()) {
        for (ByteBuffer cell : t.getCells()) {
          System.out.print(new String(cell.array()) + "\t");
        }
        System.out.println();
      }	  
    } catch(Exception e) {
      e.printStackTrace();
    }
  }
}

File: src/examples/python/SampleClient.py
#!/usr/bin/python

from gen_py.datahub import DataHub
from gen_py.datahub.constants import *
from thrift import Thrift
from thrift.protocol import TBinaryProtocol
from thrift.transport import THttpClient
from thrift.transport import TTransport

'''
@author: anant bhardwaj
@date: Oct 11, 2013

Sample Python client for DataHub
'''
username = "USERNAME"
password = "PASSWORD"
repo = "REPO"
table = "TABLE"

try:
  # Before running you will need to populate the above variables
  # and create a corresponding repo and table in your datahub acount
  print "Trying to connect to datahub with the following credentials..."
  print "\tu: " +username + "\tp: " + password
  print "\trepo: " + repo + "\ttable:" + table
  transport = THttpClient.THttpClient('http://datahub.csail.mit.edu/service')
  transport = TTransport.TBufferedTransport(transport)
  protocol = TBinaryProtocol.TBinaryProtocol(transport)
  client = DataHub.Client(protocol)
  
  print "Version: %s" %(client.get_version())
  
  # open connection
  con_params = ConnectionParams(user=username, password=password)
  con = client.open_connection(con_params=con_params)


  # execute a query
  query = "select * from " + username + "." + repo + "." + table
  print "\nTrying to execute the followin query:\t" + query + "..."

  res  = client.execute_sql(
      con=con,
      query=query,
      query_params=None)

  # print field names
  print "\t".join(res.field_names)
  
  # print tuples
  for t in res.tuples:
    print "\t".join(t.cells)

except Exception, e:
    print 'Something went wrong : %s' % (e)
